flink-clients/src/main/java/org/apache/flink/client/program/MiniClusterClient.java

```diff
@@ -74,7 +75,7 @@ public Configuration getFlinkConfiguration() {
     }
 
     @Override
-    public CompletableFuture<JobID> submitJob(@Nonnull JobGraph jobGraph) {
+    public CompletableFuture<JobID> submitJob(@Nonnull JobGraph jobGraph, Pipeline pipeline) {
```

Comment:
We discussed this with @HuangZhenQiu offline and we should remove this api change and the lineage logic will be applied on the future instead